containertest/config.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containertest
+
+import (
+	"log"
+)
+
+// This file implements the "functional options" pattern.
+
+type config struct {
+	killAfterSec   int // This is in integer seconds because that's what Docker takes.
+	service        Service
+	progressLogger Logger
+}
+
+// Logger allows the caller to optionally provide a custom logger for printing status updates about
+// service startup progress. The default is to use the go "log" package.
+type Logger interface {
+	Printf(fmtStr string, args ...any)
+}
```

Comment:
Since this is intended to be used by the `testing` package, I'm not sure this is necessary. I wonder if we should accept `testing.TB` and call `t.Log/t.Logf` instead.

Reply:
That may be better, though wouldn't it require some breaking changes to the `mysqltest` library?

Reply:
What breaking changes do you envision? We can keep the same API (even if we don't use the given logger)

Reply:
Upon further inspection, it appears that WithLogger() is never used, @drevell gave his blessing to remove it, so I will do so.

Reply:
Does it make sense to require a testing.TB when calling MustStart?

Reply:
I don't think so, because `MustStart` is intended to be called in `TestMain`, right?

Reply:
Ooof, good point. Should I change it back?

Reply:
I've changed it to return an error, so could be started within a test rather than just in TestMain. It accepts an interface which is a subset of `testing.TB`.

As a bonus, you could get isolated containers for tests that need them (especially for containers which start quickly)